File: src/com/app/test/anim/AnimActivity.java
package com.app.test.anim;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.DecelerateInterpolator;
import android.widget.ImageView;

import com.app.test.R;
import com.app.test.utils.DisplayUtil;
import com.app.test.view.TeachTalkTogetherView;

/**
 *
 */

public class AnimActivity extends Activity {
    protected ImageView ivIsVip;
    protected ImageView toOpenVipImg;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setContentView(R.layout.activity_anim);

        initView();

        initLineHeadAnima();

        ((TeachTalkTogetherView) findViewById(R.id.teach)).startAnima();
    }

    private void initLineHeadAnima() {
        final float offset = DisplayUtil.dip2px(this, 15);
//        final int offset=DisplayUtil.dip2px(this,15);


        final ViewGroup container = findViewById(R.id.container);

        ImageView pic0 = findViewById(R.id.pic0);
        ImageView pic1 = findViewById(R.id.pic1);
        ImageView pic2 = findViewById(R.id.pic2);
        ImageView pic3 = findViewById(R.id.pic3);
        ImageView pic4 = findViewById(R.id.pic4);


        pic0.setZ(4);
        pic1.setZ(3);
        pic2.setZ(2);
        pic3.setZ(1);
        pic4.setZ(0);


        final ValueAnimator valueAnimator = ValueAnimator.ofFloat(0f, offset);
        valueAnimator.setInterpolator(new DecelerateInterpolator());
        valueAnimator.setDuration(2000);
        valueAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                for (int i = 0; i < container.getChildCount(); i++) {

                    View chilidView = container.getChildAt(i);
                    float order = container.getChildCount() - 1 - chilidView.getZ();//0-4

                    if (order >= 1 && order <= 3) {
                        float transX = (float) animation.getAnimatedValue() + (order - 1) * offset;
                        chilidView.setTranslationX(transX);
                    }

                    if (order == 0) {
                        //缩放动画
                        chilidView.setScaleX(1.0f * ((float) animation.getAnimatedValue() / offset));
                        chilidView.setScaleY(1.0f * ((float) animation.getAnimatedValue() / offset));
                    } else if (order == 3) {
                        //缩放动画
//                        chilidView.setScaleX(1.0f * (1.0f - (float)animation.getAnimatedValue() / offset));
//                        chilidView.setScaleY(1.0f * (1.0f - (float)animation.getAnimatedValue() / offset));
                    }

                    if (animation.getCurrentPlayTime() == 3000) {
                        //动画结束
                        if (order == 3) {
                            chilidView.setScaleX(0);
                            chilidView.setScaleY(0);
                            chilidView.setZ(4);
                            chilidView.setTranslationX(0);
                        }

                        if (order == 2 || order == 1) {
                            chilidView.setZ(chilidView.getZ() - 1);
                        }
                    }
                }
            }
        });


        findViewById(R.id.bt).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                valueAnimator.setRepeatCount(ValueAnimator.INFINITE);
                valueAnimator.start();
            }
        });

    }

    private void initView() {
        ivIsVip = (ImageView) findViewById(R.id.iv_is_vip);
        toOpenVipImg = (ImageView) findViewById(R.id.to_open_vip_img);
        ObjectAnimator animator1 = ObjectAnimator.ofFloat(toOpenVipImg, "translationY", 0F, -50F, 0F);
        animator1.setDuration(1500);
        animator1.setRepeatCount(-1);
        animator1.start();

        AnimatorSet set = new AnimatorSet();
        set.play(animator1);
        set.start();

    }


}

File: app/src/main/java/com/app/test/anim/AnimActivity.kt
package com.app.test.anim

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import android.app.Activity
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.view.animation.DecelerateInterpolator
import android.widget.ImageView
import com.app.test.R
import com.app.test.util.DensityUtil
import com.app.test.view.TeachTalkTogetherView

/**
 *
 */
class AnimActivity : Activity() {
    private lateinit var ivIsVip: ImageView
    private lateinit var toOpenVipImg: ImageView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        super.setContentView(R.layout.activity_anim)
        initView()
        initLineHeadAnima()
        (findViewById<View>(R.id.teach) as TeachTalkTogetherView).startAnima()
    }

    private fun initLineHeadAnima() {
        val offset = DensityUtil.dp2px(15).toFloat()
        val container = findViewById<ViewGroup>(R.id.container)
        val pic0 = findViewById<ImageView>(R.id.pic0)
        val pic1 = findViewById<ImageView>(R.id.pic1)
        val pic2 = findViewById<ImageView>(R.id.pic2)
        val pic3 = findViewById<ImageView>(R.id.pic3)
        val pic4 = findViewById<ImageView>(R.id.pic4)
        pic0.z = 4f
        pic1.z = 3f
        pic2.z = 2f
        pic3.z = 1f
        pic4.z = 0f
        val valueAnimator = ValueAnimator.ofFloat(0f, offset)
        valueAnimator.interpolator = DecelerateInterpolator()
        valueAnimator.duration = 2000
        valueAnimator.addUpdateListener { animation ->
            for (i in 0 until container.childCount) {
                val chilidView = container.getChildAt(i)
                val order = container.childCount - 1 - chilidView.z //0-4
                if (order in 1.0..3.0) {
                    val transX = animation.animatedValue as Float + (order - 1) * offset
                    chilidView.translationX = transX
                }
                if (order == 0f) {
                    //缩放动画
                    chilidView.scaleX = 1.0f * (animation.animatedValue as Float / offset)
                    chilidView.scaleY = 1.0f * (animation.animatedValue as Float / offset)
                } else if (order == 3f) {
                    //缩放动画
//                        chilidView.setScaleX(1.0f * (1.0f - (float)animation.getAnimatedValue() / offset));
//                        chilidView.setScaleY(1.0f * (1.0f - (float)animation.getAnimatedValue() / offset));
                }
                if (animation.currentPlayTime == 3000L) {
                    //动画结束
                    if (order == 3f) {
                        chilidView.scaleX = 0f
                        chilidView.scaleY = 0f
                        chilidView.z = 4f
                        chilidView.translationX = 0f
                    }
                    if (order == 2f || order == 1f) {
                        chilidView.z = chilidView.z - 1
                    }
                }
            }
        }
        findViewById<View>(R.id.bt).setOnClickListener {
            valueAnimator.repeatCount = ValueAnimator.INFINITE
            valueAnimator.start()
        }
    }

    private fun initView() {
        ivIsVip = findViewById<View>(R.id.iv_is_vip) as ImageView
        toOpenVipImg = findViewById<View>(R.id.to_open_vip_img) as ImageView
        val animator1 = ObjectAnimator.ofFloat(toOpenVipImg, "translationY", 0f, -50f, 0f)
        animator1.duration = 1500
        animator1.repeatCount = -1
        animator1.start()
        val set = AnimatorSet()
        set.play(animator1)
        set.start()
    }
}